### PR TITLE
Add 'trim_all' option to re:split/3

### PR DESCRIPTION
**Pull request by capflam**
This option can be set to remove _ALL_ empty parts of the result of a call to re:split/3.
This pull request is related to #517

### CODE REVIEW

File: lib/stdlib/test/re_SUITE_data/testoutput1 [hunk 1]
```diff
@@ -22,6 +22,10 @@ No match
     What do you know about THE QUICK BROWN FOX?
  0: THE QUICK BROWN FOX
 
+/\s/
+    \x20\x20\x20A\x20\x20\x20B\x20\x20\x20C\x20\x20\x20\x20\x20\x20\x20\x20\x20
+ 0: \x20
+
```
Comment (bjorng) [inline lib/stdlib/test/re_SUITE_data/testoutput1]: Place these lines at the end of the file. That will ensure that re_testoutput1_replacement_test.erl will not change.

File: lib/stdlib/test/run_pcre_tests.erl [hunk 2]
```diff
@@ -1063,6 +1063,21 @@ dumponesplit(F,{RE,Line,O,TS}) ->
...
 			    dsafe2(safe(RE)),
 			    NO++[{parts,2}]]),
+
+		 %% Use a special separator here ('§') to avoid conflicts
+		 io:format(F,"perl -e '$x = join(\"§\",split(/~s/~s,\"~s\",-1)); "
+			   "$x =~~ s/(^§+|§+$)//g; $x =~~ s/§§+/§/g; $x =~~ s/§/:/g; "
+			   "$x =~~ s/\\\\/\\\\\\\\/g; $x =~~ s/\\\"/\\\\\"/g; "
+			   "print \"    <<\\\"$x\\\">> = "
+			   "iolist_to_binary(join(re:split(\\\"~s\\\","
+			   "\\\"~s\\\",~p))), \\n\";'~n",
```
Comment (bjorng) [inline lib/stdlib/test/run_pcre_tests.erl]: Remove the space before "\\n" (I know that the other io:formats have space in that position, but there is no need to add redundant spaces to new lines).

**Comment by bjorng** [comment]
Looks good, except for a few details. The main thing is that new option must be documented.

Please rebase this branch on the master branch (we usually don't add new features to the maint branch).

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by bjorng** [comment]
Closing this because of no response. Submit a new pull request if/when you have done the proposed changes.

**Pull request closed without merging**